q.js

```diff
@@ -1645,6 +1645,9 @@ Q["finally"] = function (object, callback) {
 
 Promise.prototype.fin = // XXX legacy
 Promise.prototype["finally"] = function (callback) {
+    if (callback === undefined) {
```

Comment:
`if (!callback)` would suffice; `if (typeof callback.apply !== "function")` would be strong. If we do this here, we should do it for all methods.